watertap/tools/parameter_sweep.py

```diff
@@ -319,108 +375,288 @@ def _interp_nan_values(global_values, global_results):
+        output_dict["sweep_params"][var_str] =  _create_component_output_skeleton(var, num_samples)
 
-        model : A Pyomo ConcreteModel containing a watertap flowsheet, for best 
-                results it should be initialized before being passed to this
-                function.
+    # 2. lets deal with the output variables from the pyomo model
+    for var in variables_in_activated_equalities_set(model.fs):
+        var_str = var.name
+        output_dict["outputs"][var_str] = _create_component_output_skeleton(var, num_samples)
 
-        sweep_params: A dictionary containing the values to vary with the format
-                      ``sweep_params['Short/Pretty-print Name'] =
-                      (model.fs.variable_or_param[index], lower_limit, upper_limit, num_samples)``.
-                      A uniform number of samples ``num_samples`` will be take between
-                      the ``lower_limit`` and ``upper_limit``.
+    # 3. lets deal with expressions
+    for expr in expressions_set(model.fs):
+        expr_str = expr.name
+        output_dict["outputs"][expr_str] = _create_component_output_skeleton(var, num_samples)
 
-        outputs : A dictionary containing "short names" as keys and and Pyomo objects
-                  on ``model`` whose values to report as values. E.g.,
-                  ``outputs['Short/Pretty-print Name'] = model.fs.variable_or_expression_to_report``.
+    # 4. Lets deal with objectives
+    for obj in total_objectives_set(model):
+        obj_str = obj.name
+        output_dict["outputs"][obj_str] = _create_component_output_skeleton(var, num_samples)
 
-        results_file (optional) : The path and file name where the results are to be saved;
-                                   subdirectories will be created as needed.
+    return output_dict
 
-        optimize_function (optional) : A user-defined function to perform the optimization of flowsheet
-                                       ``model`` and loads the results back into ``model``. The first
-                                       argument of this function is ``model``\. The default uses the
-                                       default IDAES solver, raising an exception if the termination
-                                       condition is not optimal.
+# ================================================================
 
-        optimize_kwargs (optional) : Dictionary of kwargs to pass into every call to
-                                     ``optimize_function``. The first arg will always be ``model``,
-                                     e.g., ``optimize_function(model, **optimize_kwargs)``. The default
-                                     uses no kwargs.
+def _create_component_output_skeleton(component, num_samples):
+    # TODO: Revisit thie variable "component" name
+
+    comp_dict = {}
+    attr_list  = dir(component)
+    comp_dict["value"] = np.zeros(num_samples, dtype=np.float)
+    if 'lb' in attr_list:
```

Comment:
Why not just use `hasattr`?
```suggestion
    if hasattr(component, 'lb'):
```

Reply:
Incorporated in a later commit.